src/services/request-service.ts

```diff
@@ -40,19 +42,29 @@ export class RequestService {
   constructor(
     config: Config,
     private readonly requestRepository: RequestRepository,
+    private readonly replicationRequestRepository: IReplicationRequestRepository,
     private readonly requestPresentationService: RequestPresentationService,
     private readonly metadataService: IMetadataService,
     private readonly validationQueueService: IQueueProducerService<RequestQMessage>
   ) {
     this.publishToQueue = Boolean(config.queue.sqsQueueUrl)
   }
 
+  /**
+   * Finds a request by CID from read replica first if not found there then from main db
+   * @param cid The CID of the request.
+   * @returns The request.
+   */
   async getStatusForCid(cid: CID): Promise<OutputOf<typeof CASResponse> | { error: string }> {
-    const request = await this.requestRepository.findByCid(cid)
+    let request = await this.replicationRequestRepository.findByCid(cid)
     if (!request) {
-      throw new RequestDoesNotExistError(cid)
+      request = await this.requestRepository.findByCid(cid)
+      logger.debug(`Request not found in replica db for ${cid}, fetching from main_db`)
+      Metrics.count(METRIC_NAMES.REPLICA_DB_REQUEST_NOT_FOUND, 1)
```

Comment:
Do we want to move this above line 61 (if line 61 errors, the metric won't be counted)

Reply:
Same for below